fs/fs_heap.c

```diff
@@ -0,0 +1,66 @@
+/****************************************************************************
+ * fs/fs_heap.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include "fs_heap.h"
+
+#if CONFIG_FS_HEAPSIZE > 0
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct mm_heap_s *g_fs_heap = NULL;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void fs_heap_initialize(void)
+{
+  FAR void *buf = kmm_malloc(CONFIG_FS_HEAPSIZE);
+  DEBUGASSERT(buf != NULL);
```

Comment:
why not define to static array to avoid alloc fail?

Reply:
static and kmm maybe in different link section, 
in this case put big array in .bss we should do link script modification, maybe use kmalloc prefer.

as we can always know if the CONFIG_FS_HEAPSIZE is bigger than heap can accept when bootup. keep in heap does not have much side effect.

Reply:
If CONFIG_FS_HEAPSIZE is defined to be larger than the size of the heap, then it will be asserted at runtime.
If move the fs buffer to the heap, we could get the ld error in the linking phase.

Reply:
> in this case put big array in .bss we should do link script modification, maybe use kmalloc prefer.

we do not need to modify the link script, it will be placed in the bss area by default when linking, unless you have a specific section requirement, then using kmm will not meet this requirement

Reply:
It is not good to define the heap buffer as global array since the boot code will waste time to zero this huge array, but which isn't require for heap manager at all.

Reply:
I will add static buffer support later, which will help devices detect issue in link phase, rather than after the system starts.